tslib/src/azure/storage-helper.ts

```diff
@@ -51,10 +49,10 @@ export class AsyncTableService extends az.TableService implements IAsyncTableSer
     })
   }
 
-  queryEntitiesAsync (table: string, tableQuery: az.TableQuery, currentToken: az.TableService.TableContinuationToken): Promise<Error | any> {
+  async queryEntitiesAsync (table: string, tableQuery: az.TableQuery, currentToken: az.TableService.TableContinuationToken): Promise<Error | any> {
```

Comment:
`async` is not required as it does not actually use an await.  Just returning a promise is enough for it to be await'd on.

Reply:
yes i'm aware of this, i usually omit the `async` keyword too.  This is the reasoning behind the rule - https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/promise-function-async.md